realm/realm-library/src/main/cpp/objectserver_shared.hpp

```diff
@@ -60,7 +61,19 @@ class JniSession {
                 coordinator->wake_up_notifier_worker();
             }
         };
-        auto error_handler = [weak_session_ref](std::error_code error_code, bool /*is_fatal*/, const std::string message) {
+        auto error_handler = [weak_session_ref](std::error_code error_code, bool is_fatal, const std::string message) {
+            if (error_code.category() != realm::sync::protocol_error_category() ||
+                    error_code.category() != realm::sync::client_error_category()) {
+                // FIXME: Consider below when moving to the OS sync manager.
```

Comment:
OS just returns https://github.com/realm/realm-object-store/pull/303/files#diff-f3165f1bbea4d4c8abf9c03a4b4e86f9R385 when this happens.

If it is an fatal error, we need to do something with it. But we are not doing it before this PR, so I think we should handle it in another PR or in the OS sync manager PR.